Examples/Shared/SplashViewController.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+import UIKit
+
+final class SplashViewController: UIViewController {
+    private let userInfo: UserInfo
+    private let token: Token
+    private let completionHandler: () -> Void
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+    
+    init(
+        userInfo: UserInfo = .init(id: "user-1"),
+        token: Token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiY2lsdmlhIn0.jHi2vjKoF02P9lOog0kDVhsIrGFjuWJqZelX5capR30",
+        completionHandler: @escaping () -> Void
+    ) {
+        self.userInfo = userInfo
+        self.token = token
+        self.completionHandler = completionHandler
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .systemBackground
+        let activityIndicatorView = UIActivityIndicatorView(style: .large)
+        activityIndicatorView.startAnimating()
+        activityIndicatorView.center = view.center
+        view.addSubview(activityIndicatorView)
+        
+        connectUser()
+    }
+    
+    private func connectUser() {
+        ChatClient.shared.connectUser(
+            userInfo: userInfo,
+            token: token
+        ) { [weak self] error in
+            DispatchQueue.main.async {
+                if let error {
+                    self?.presentAlert(with: error)
+                } else {
+                    self?.completionHandler()
+                }
+            }
+        }
+    }
+    
+    private func presentAlert(
+        with error: Error
+    ) {
+        let alertController = UIAlertController(
+            title: "Error",
+            message: error.localizedDescription,
+            preferredStyle: .alert
+        )
+        
+        alertController.addAction(.init(
```

Comment:
Just one more thing here: If the connection fails, and the user cancels, we would end up with endless spinner. Maybe just switch the spinner to a label with an error message if that happens?

Reply:
Updated with one step further. Added a message and a button to restart the connectUser flow. Let me know your thoughts

Reply:
great, even better 👍